src/serverExplorer.ts

```diff
@@ -96,10 +96,13 @@ export class ServersViewTreeDataProvider implements TreeDataProvider<Protocol.Se
             canSelectFolders: true,
             openLabel: 'Select desired server location'
         }).then(folders => {
-            if (folders && folders.length === 1) {
-                return this.client.findServerBeans(folders[0].fsPath);
+                if (!folders) {
+                    Promise.reject();
```

Comment:
`Promise.reject()` creates rejected Promise. If it is not returned from then callback it does not affect promise chain and it would just keep going from one then callback call to another.
The fix works the same as original code.  
If `folders` callback parameter is null/undefined, in case folder selection dialog is canceled, it will create rejected promise and then return `undefined` from callback. This 'undefined` value then going to be passed through `serverBeans` parameter to next callback which would do nothing in this case and just pass `undefined` to next callback as data parameter.